packages/analytics-js/src/components/capabilitiesManager/detection/dom.ts

```diff
@@ -4,16 +4,21 @@ const isDatasetAvailable = (): boolean => {
   return testElement.dataset ? testElement.dataset.aB === 'c' : false;
 };
 
+// This checks go hand to hand with the REQUIRED_POLYFILL_LIST
```

Comment:
Can we define these together to make this less error-prone?

```
const REQUIRED_POLYFILL_LIST: Record<string,() => boolean> = {
  'URLSearchParams': () => !String.prototype.endsWith
  'URL': () => !window.URL
  'Element.prototype.dataset': () => !isDatasetAvailable()
};

const isLegacyJSEngine = (): boolean => Object.values(REQUIRED_POLYFILL_LIST).some(x => x());

const POLYFILL_URL = `https://polyfill.io/v3/polyfill.min.js?features=${Object.keys(REQUIRED_POLYFILL_LIST).join(
  '%2C',
)}`;
```

Reply:
on it

Reply:
done